File: engine/src/main/java/ru/vertuos/engine/map/object/CollisionableGameObject.java
package ru.vertuos.engine.map.object;

import lombok.Getter;
import ru.vertuos.engine.world.object.DynamicGameObject;
import ru.vertuos.engine.world.object.GameObject;

@Getter
public abstract class CollisionableGameObject extends GameObject {

    public CollisionableGameObject(float positionX, float positionY, float width, float height) {
        this.positionX = positionX;
        this.positionY = positionY;
        this.width = width;
        this.height = height;
    }

    public abstract void onCollide(DynamicGameObject dynamic);
}

File: engine/src/main/java/ru/vertuos/engine/map/CollisionableGameObject.kt
package ru.vertuos.engine.map

import ru.vertuos.engine.math.Vector2f
import ru.vertuos.engine.world.obj.DynamicGameObject
import ru.vertuos.engine.world.obj.GameObject

abstract class CollisionableGameObject(
    position: Vector2f,
    size: Vector2f
) : GameObject() {

    init {
        this.position = position
        this.size = size
    }

    abstract fun onCollide(obj: DynamicGameObject)
}
